OHIF/Viewers
# feat(studyPrefetcher): Study Prefetcher

Added support for a Study Prefetcher responsible for loading the next display sets. The following settings are available:

```javascript
  studyPrefetcher: {
    /* Enable/disable study prefetching service (default: false) */
    enabled: true,
    /* Number of displaysets to be prefetched  (default: 2)*/
    displaySetCount: 2,
    /**
     * Max number of concurrent prefetch requests (default: 10)
     * High numbers may impact on the time to load a new dropped series because
     * the browser will be busy with all prefetching requests. As soon as the
     * prefetch requests get fulfilled the new ones from the new dropped series
     * are sent to the server.
     *
     * TODO: abort all prefetch requests when a new series is loaded on a viewport.
     * (need to add support for `AbortController` on Cornerstone)
     * */
    maxNumPrefetchRequests: 10,
    /* Display sets loading order (closest (deafult), downward or upward) */
    order: 'closest',
  },
```

### Testing

- Disable the StudyPrefetcher and make sure it does not prefetch the next display sets (1).
- Enable the StudyPrefetcher and make sure the next display sets are loaded (1) based on the configuration provided.
- Load a new series, change the viewport layout (eg: 1x1 to 1x2), change the active viewport and see if the displaysets are loaded as expected based on the display set loaded on the active viewport (eg: making viewport 2 as active should stop loading the series based on viewport 1 and start loading the ones related to viewport 2).

(1) The progress is displayed while a series is being prefetched underneath its thumbnail (eg: "P: 47%") but **the design will change soon** (the text will be replaced by a progress bar).
